astropy/units/function/core.py

```diff
@@ -267,6 +267,13 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
+    def __rlshift__(self, m):
+        """Unit converstion operator `<<`"""
+        try:
+            return FunctionQuantity(m, self, copy=False, subok=True)
+        except Exception:
```

Comment:
Do we need this `except` block? This makes it harder to debug if things go wrong. @mhvk knows best though.

Reply:
Without looking in more detail at `m` we cannot know whether it is in fact possible to construct a `FunctionQuantity`. Returning `NotImplemented` will end of raising `TypeError`, which is arguably the right answer.